fix(gui): add RTL support to delete-confirmation-prompt

pishi-ai opened the pull request:
This commit adds RTL (Right-to-Left) language support to the delete-confirmation-prompt component in scratch-gui.

### Resolves

In RTL languages, when multiple sprites are opened, the confirmation prompt extends beyond the window boundary. This PR fixes that issue by properly handling RTL layout.

### Proposed Changes

- Updated layout and styles to support RTL rendering for the delete-confirmation-prompt
- Adjusted alignment logic and directional styling
- Ensured icon and button positioning respects RTL direction
- Modified 6 existing files
- Added 1 new file

### Reason for Changes

RTL language support is essential for accessibility and internationalization. Without proper handling, the confirmation prompt becomes unusable in RTL contexts. This update ensures consistent behavior and layout across both LTR and RTL languages.

### Test Coverage

- Manually tested the delete-confirmation-prompt in RTL and LTR languages
- Verified the prompt displays fully within the window and aligns correctly
- Confirmed no regression in LTR behavior

Commit:
fix(gui): rtl support for delete-confirmation-prompt

This commit adds RTL (Right-to-Left) support for the delete-confirmation-prompt.
In RTL languages, when multiple sprites are opened, the delete section of the
confirmation prompt extended beyond the window boundary. This update fixes the
layout so it displays correctly in RTL as well as LTR.

pishi-ai commented:
I have read the CLA Document and I hereby sign the CLA

cwillisf commented:
@pishi-ai, thanks for your help! I really appreciate the testing notes, too.

cwillisf commented:
@KManolov3, do you have time to review this? If not, let me know.
I took a quick look: it seems generally reasonable, but you're the expert here :)

KManolov3 commented:
This looks good! I think we should try to get it merged for one of the upcoming releases.

cwillisf commented:
Sorry for the bother, @pishi-ai, but could you merge from our `develop` branch into this PR's branch?

It looks like something went wrong and prevented the normal build-and-test steps from running for this PR. I've adjusted our settings, so those steps should run if you make any changes to this PR. Merging from our `develop` branch seems like the most innocuous change available :)

pishi-ai commented:
recheck

pishi-ai commented:
I have read the CLA Document and I hereby sign the CLA

pishi-ai commented:
The failing tests don’t seem related to this change. Could a maintainer confirm?
@cwillisf

pishi-ai committed:
fix(gui): add RTL support to delete-confirmation-prompt

This commit adds RTL (Right-to-Left) support for the delete-confirmation-prompt.
In RTL languages, when multiple sprites are opened, the delete section of the
confirmation prompt extended beyond the window boundary. This update fixes the
layout so it displays correctly in RTL as well as LTR.

pishi-ai committed:
fix(gui): add RTL support to delete-confirmation-prompt

This commit adds RTL (Right-to-Left) support for the delete-confirmation-prompt.
In RTL languages, when multiple sprites are opened, the delete section of the
confirmation prompt extended beyond the window boundary. This update fixes the
layout so it displays correctly in RTL as well as LTR.

cwillisf commented:
It looks like the test failures are related. I guess sometimes `state.locales` is undefined in `state.locales.isRtl`.

```raw
Summary of all failing tests
FAIL test/unit/containers/sprite-selector-item.test.jsx
  ● SpriteSelectorItem Container › should delete the sprite, when called without `withDeleteConfirmation`

    TypeError: Cannot read properties of undefined (reading 'isRtl')
      
      at Function.mapStateToProps [as mapToProps] (src/containers/sprite-selector-item.jsx:208:26)
      at mapToPropsProxy (../../node_modules/react-redux/lib/connect/wrapMapToProps.js:53:46)
```

```raw
  ● SpriteSelectorItem Container › should initiate sprite deletion, when called `withDeleteConfirmation`

    TypeError: Cannot read properties of undefined (reading 'isRtl')
      
      at Function.mapStateToProps [as mapToProps] (src/containers/sprite-selector-item.jsx:208:26)
      at mapToPropsProxy (../../node_modules/react-redux/lib/connect/wrapMapToProps.js:53:46)
```

cwillisf commented:
I just noticed that you can't see the errors, which certainly doesn't help. I'll try to see if I can fix that in a safe way...

pishi-ai committed:
fix(gui): add missing locales state to sprite-selector-item test

fix: resolve undefined locales.isRtl error in sprite-selector-item tests

Add missing locales state to test mock store and fix DeleteConfirmationPrompt mock path.
The component's mapStateToProps was trying to access state.locales.isRtl which was undefined.

pishi-ai committed:
fix(gui): add RTL support to delete-confirmation-prompt

This commit adds RTL (Right-to-Left) support for the delete-confirmation-prompt.
In RTL languages, when multiple sprites are opened, the delete section of the
confirmation prompt extended beyond the window boundary. This update fixes the
layout so it displays correctly in RTL as well as LTR.

pishi-ai commented:
> I just noticed that you can't see the errors, which certainly doesn't help. I'll try to see if I can fix that in a safe way...

Thanks for the feedback, @cwillisf! I've addressed your comments and fixed the errors. It should be ready for another review now.

pishi-ai committed:
fix(gui): add RTL support to delete-confirmation-prompt

This commit adds RTL (Right-to-Left) support for the delete-confirmation-prompt.
In RTL languages, when multiple sprites are opened, the delete section of the
confirmation prompt extended beyond the window boundary. This update fixes the
layout so it displays correctly in RTL as well as LTR.

pishi-ai committed:
fix(gui): add RTL support to delete-confirmation-prompt

This commit adds RTL (Right-to-Left) support for the delete-confirmation-prompt.

In RTL languages, when multiple sprites are opened, the delete section of the

confirmation prompt extended beyond the window boundary. This update fixes the

layout so it displays correctly in RTL as well as LTR.

pishi-ai committed:
test(gui): add locales state to sprite-selector-item test

Adds missing locales mock state required for components that use RTL support.

Fixes test setup to match actual Redux state shape.
